Develop sequencing

ilakshmank opened the pull request:
tested and validated start up sequencing and calibration. 
Added macros to turn off audible alarms for debugging.

Commit:
power: create initial power on/power off flow

Signed-off-by: Rosen, Michael R <michael.r.rosen@intel.com>

Commit:
calibration: implement ui-based calibration

. Both sensors are now calibrated via UI command instead
  of automatically at boot

Signed-off-by: Rosen, Michael R <michael.r.rosen@intel.com>

Commit:
sys: fix sys hal power off

Fix sys HAL power off API to spin until power down

Signed-off-by: Rosen, Michael R <michael.r.rosen@intel.com>

Commit:
main: add 5s delay before actually cutting power

To allow the UI-CU to shutdown, add a 5s power off
delay between receiving the power off message and
actually cutting off power to the system.

Signed-off-by: Rosen, Michael R <michael.r.rosen@intel.com>

Commit:
main: add sys hal initialization

Signed-off-by: Rosen, Michael R <michael.r.rosen@intel.com>

ilakshmank committed:
Signed-off-by: Lakshman Krishnamurthy <ilakshmank@gmail.com>

 Changes to be committed:
	modified:   src/hal/sys.cpp
	modified:   src/hal/sys.h
	modified:   src/main.cpp

Fixed power on reboot bug and power off assert

ilakshmank committed:
Signed-off-by: Lakshman Krishnamurthy <ilakshmank@gmail.com>

 Changes to be committed:
	modified:   main.cpp
	modified:   modules/sensors.cpp

Fixed power off alarm to AMMI spec
Fixed no sesnor alarms to until sensoes are calibrated

mrrosen commented on src/modules/sensors.cpp:
```diff
@@ -325,7 +325,8 @@ static PT_THREAD(sensorsPressureThreadMain(struct pt* pt))
     // Alarms
-    if ((control.state != CONTROL_STATE_HOME) && (control.state != CONTROL_STATE_IDLE)) {
+    if ((control.state != CONTROL_STATE_HOME) && (control.state != CONTROL_STATE_IDLE)
+	&& (sensors.calibrated == SENSORS_ALL_CALIBRATED)) {
```
This should really only check if the pressure sensor is calibrated. and there should be another one of these for the airflow sensor

```suggestion
	&& (sensors.calibrated & SENSORS_PRESSURE_CALIBRATED)) {
```

mrrosen commented on src/main.cpp:
```diff
@@ -210,15 +210,21 @@ void mainLoop(void)
+	 } else if(timerHalRun(&powerTimer) == HAL_TIMEOUT) {
+	   LOG_PRINT(INFO, "Canceling  power off");
+	   alarmSuppress(&powerOffAlarm);
+	   powerOffStep = POWEROFF_STEP0;
```
The timer was originally indented to enable "peaceful" software shutdown in the event of a UI hang. If we dont really care about this (since if the UI is stuck, theres always the hard power off), we can get rid of the timer and have the controller ignore the button entirely. That way, if the UI never issues a powerOff command, the controller will just ignore it, making the UI in control of the power off sequence initialization.

ilakshmank replied:
The timer I added to turn off the alarm. The idea is as soon as the power button is pressed you don't continue pressing it for 12 seconds in error. This means that the sound should also come in the vent is in active ventilation mode.

mrrosen replied:
The previous implementation was implemented in accordance with the diagrams Dawn created, though I did add that timer to catch UI hangs when I shouldnt have according to these flows. If you want to change the behavior, you should probably make sure the flow is clear:
Flow if not ventilating:
https://docs.google.com/drawings/d/18HMhex4woUm92CJvpRbYs7WUHl_WF7-8fJD0X3kRqfE/edit
Flow if ventilating:
https://docs.google.com/drawings/d/1NMcsxk5UIIg3xlkM4I24ktSFHBFxSSZU0pOrE6gMK1k/edit

ilakshmank replied:
Ok. I agree with the previous implementation - what I did had a bug.

   case POWEROFF_STEP0:
        if (comm.powerOff) || (sysHalPowerButtonAsserted()) {
          LOG_PRINT(INFO, "Power button pushed, preparing to power off...");
          timerHalBegin(&powerTimer, POWEROFF_TIMEOUT, false);
	  alarmSet(&powerOffAlarm);
          powerOffStep = POWEROFF_STEP1;
        }

is OK.  I did change the diagram do reflect what I actually happening.

https://docs.google.com/drawings/d/18HMhex4woUm92CJvpRbYs7WUHl_WF7-8fJD0X3kRqfE/edit

ilakshmank replied:
OK. I updated the diagrams and the code has a common path for  vent and non-vent mode.

mrrosen reviewed (changes requested):
@ilakshmank just a few changes

ilakshmank committed:
Update src/modules/sensors.cpp

Co-authored-by: Rosen, Michael R <michael.r.rosen@intel.com>

ilakshmank committed:
Signed-off-by: Lakshman Krishnamurthy <ilakshmank@gmail.com>

comman path for vent and no vent case during shutdown
alarm always comes on for 10 seconds

ilakshmank committed:
Merge branch 'develop-sequencing' of https://github.com/OVVE/software-controller into develop-sequencing

ilakshmank committed:
Signed-off-by: Lakshman Krishnamurthy <ilakshmank@gmail.com>

removed   POWEROFF_STEP 4 and eliminated STEP2 contents
